Search Users, Search Moods, Search My Moods (Previously #249)

TahaKamil opened the pull request:
## Summary
 - Implemented Search Users on Home Feed
 - Search Moods on Home Feed
 - Search My Moods on Mood History page

Resolves #103 
Resolves: #104 
Resolves: #105  
Resolves: #207 

## Testing

- How should a reviewer test this feature, fix, or UI?
     - Try Searching for users, moods in home page
     - Try searching for my moods on mood history page
- Are there any specific commands, steps, or credentials needed?
     - N?A

## Merge Instructions

- Any considerations that other PR's may need to take into note? 
     - N/A
- Should the reviewer delete the branch on merge? 
     - Yes

---

**Checklist**
- [x] Are all relevant issues referenced by this PR?
- [x] Does the app still build (locally)?
- [x] Do all tests pass?
- [x] Does the feature work as expected?
- [x] Does this feature preserve the app’s existing stability?  
- [x] Is the description of the changes correct/present?
- [x] Have new tests been created or existing tests updated as needed?

---

adityadipakpatel committed:
some code updates

adityadipakpatel committed:
more updates

adityadipakpatel committed:
build errors fix

adityadipakpatel committed:
home feed to other user profile page for search users

adityadipakpatel committed:
Search Users working

adityadipakpatel committed:
Search Moods almost done

adityadipakpatel committed:
change "Reason" to "Moods"

adityadipakpatel committed:
Search Moods done

adityadipakpatel committed:
change "Moods" to "Mood Reason"

adityadipakpatel committed:
Filter only by (Mood Reason) not (Mood Reason or Mood Type)

adityadipakpatel committed:
basic test

adityadipakpatel committed:
bug fixes

adityadipakpatel committed:
test runtime bugs fixed

adityadipakpatel committed:
udpated detailed tests

adityadipakpatel committed:
test build bugs fix

adityadipakpatel committed:
update to test file

adityadipakpatel committed:
add sleep

adityadipakpatel committed:
Test 1 bug at line 163

adityadipakpatel committed:
search based on reason not by emotion type in Mood History Page

adityadipakpatel committed:
Test 1 bug fixes

adityadipakpatel committed:
Test 1, 3 done

adityadipakpatel committed:
Test 2 changes

adityadipakpatel committed:
Test 2 Still Buggy

adityadipakpatel committed:
start test 2 again from scratch

adityadipakpatel committed:
Test 2 re-written

adityadipakpatel committed:
Test 2 update

adityadipakpatel committed:
small fix

adityadipakpatel committed:
remove main file

adityadipakpatel committed:
Refactor Tests into 3 individual Tests

TahaKamil commented:
@adityadipakpatel When you are available, we should discuss which/what we are merging. We have reverted the merge for now.

adityadipakpatel commented:
> @adityadipakpatel When you are available, we should discuss which/what we are merging. We have reverted the merge for now.

its the search users(home feed page), search all moods by reason(home feed page), search my moods by reason(mood history page) as stated in the title of [my original PR](https://github.com/cmput301-w25/project-kernelcrew/pull/249)

TahaKamil commented:
> > @adityadipakpatel When you are available, we should discuss which/what we are merging. We have reverted the merge for now.
> 
> 
> 
> its the search users(home feed page), search all moods by reason(home feed page), search my moods by reason(mood history page) as stated in the title of [my original PR](https://github.com/cmput301-w25/project-kernelcrew/pull/249)

Okay then, we should discard your pr since I already implemented that in mine along with the filtering. I also noticed you changed a lot of things that I need for my functions and implementations  meeded for the filtering...like the drop down menus, event listeners, etc.

siddhantgoel7 commented:
@TahaKamil in this branch the searching is complete and bug free and the filters work. Only the filters for location is not working. I would suggest you to do the location filters here or let us know what you think

siddhantgoel7 committed:
merge
